### PR TITLE
[UI] Create confirmation modal for test profile delete 

### LINKED ISSUE #3306: [UI] Performance Profiles: Updates to Cards
#### Current Behavior
Performance Profile cards have an intriguing user experience, but also has a few areas of need of improvement.

#### Implementation
1. Change Delete action behavior: present user a confirmation modal before deleting the performance profile
1. Add all profile details to the back of the card: service mesh type, QPS and Concurrent requests and custom headers, if present.
1. Move Delete and Edit buttons to back of the card.
1. Remove the "Details" button from the front of the card.
1. Move "view test results" to the front. Rename to "View Results". 
1. Activate card on hover. Provide another indication to the user that they can click on the card.
    1. Option: tilt card ever so slightly
    1. Option: lightly tint the background of the card

#### Acceptance Tests

- [x] 1. Perf profiles require user confirmation prior to being deleted.
- [ ] 2. Users can see all details of a performance profile in a read-only fashion on the back of the profile card, including custom headers if present.
- [ ] 3. Users can invoke administrative operations from the back of the card, not the front.
- [ ] 4. Users can click "View Results" to expand the card, showing test result details in tabular form.
- [ ] 5. User is provided indication that cards are clickable both by changing the pointer to "hand" and one other _subtle_ visual indicator.

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by humblenginr**
This PR
 Adds delete confirmation modal
 Moves the  edit and delete buttons to the back of the card

Signed-off-by: suod-NithishKarthik <nithishkarthik01@gmail.com>

**Description**

This PR fixes #3306

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Ayush-Khandelwal-007** [comment]
Hey, @sudo-NithishKarthik can you please make the dialog match up the present UI theme. Something is,ilar to the dialog box that appears when w remove the cluster config.

**Comment by sudo-NithishKarthik** [comment]
> Hey, @sudo-NithishKarthik can you please make the dialog match up the present UI theme. Something similar to the dialog box that appears when w remove the cluster config.

Thank you for pointing it out, i actually did not notice that there is actually a HOC called `PromptComponent` which is designed for this purpose. So, now i have used it instead of creating a new modal.


https://user-images.githubusercontent.com/75248557/124111144-9a0dca80-da86-11eb-9efe-94ccbcb86115.mov

**Review by Ayush-Khandelwal-007 — APPROVED** [review]
LGTM! 🎉

**Comment by leecalcote** [comment]
❤️ screen recordings 😄

**Comment by leecalcote** [comment]
Link to design component for modals - https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=2328%3A10

**Comment by leecalcote** [comment]
Do we now have a reusable component for this type of dialog box that we can consistently use in all UIs?

**Comment by sudo-NithishKarthik** [comment]
> Do we now have a reusable component for this type of dialog box that we can consistently use in all UIs?

Yup, we do !

File: ui/components/MesheryPerformance/PerformanceProfiles.js [hunk 1]
```diff
@@ -106,7 +110,13 @@ function PerformanceProfile({ updateProgress, enqueueSnackbar, closeSnackbar })
     );
   }
 
-  function deleteProfile(id) {
+  async function deleteProfile(id) {
+    let response = await modalRef.current.show({
+      title: "Delete confirmation",
+      subtitle: "Are you sure you want to delete this test profile. This action cannot be undone.",
```
Comment (leecalcote) [inline ui/components/MesheryPerformance/PerformanceProfiles.js:116]:
```suggestion
      subtitle: "Are you sure you want to delete this performance profile?",

```

File: ui/components/MesheryPerformance/PerformanceProfiles.js [hunk 2]
```diff
@@ -106,7 +110,13 @@ function PerformanceProfile({ updateProgress, enqueueSnackbar, closeSnackbar })
     );
   }
 
-  function deleteProfile(id) {
+  async function deleteProfile(id) {
+    let response = await modalRef.current.show({
+      title: "Delete confirmation",
```
Comment (leecalcote) [inline ui/components/MesheryPerformance/PerformanceProfiles.js:115]:
```suggestion
      title: "Delete Performance Profile?",

```

File: ui/components/MesheryPerformance/PerformanceProfiles.js [hunk 3]
```diff
@@ -106,7 +110,13 @@ function PerformanceProfile({ updateProgress, enqueueSnackbar, closeSnackbar })
     );
   }
 
-  function deleteProfile(id) {
+  async function deleteProfile(id) {
+    let response = await modalRef.current.show({
+      title: "Delete confirmation",
+      subtitle: "Are you sure you want to delete this test profile. This action cannot be undone.",
+      options: ["yes", "no"],
```
Comment (leecalcote) [inline ui/components/MesheryPerformance/PerformanceProfiles.js:117]:
```suggestion
      options: ["YES", "NO"],

```

**Review by leecalcote — APPROVED** [review]
This is great. 👍

**Pull request merged**